### PR TITLE
samples: wifi: sta: Use wifi_cred library

### PR DESCRIPTION
**Pull request by D-Triveni**
Add changes to use wifi_credentials library.

### CODE REVIEW

File: samples/wifi/sta/src/main.c [hunk 1]
```diff
@@ -125,6 +128,7 @@ static int cmd_wifi_status(void)
 		LOG_INF("MFP: %s", wifi_mfp_txt(status.mfp));
 		LOG_INF("RSSI: %d", status.rssi);
 	}
+
```
Comment (krish2718) [inline samples/wifi/sta/src/main.c:131]: unnecessary whitespace

File: samples/wifi/sta/src/main.c [hunk 2]
```diff
@@ -24,6 +24,9 @@ LOG_MODULE_REGISTER(sta, CONFIG_LOG_DEFAULT_LEVEL);
 #include <zephyr/net/net_event.h>
 #include <zephyr/drivers/gpio.h>
 
+#include <net/wifi_credentials.h>
```
Comment (krish2718) [inline samples/wifi/sta/src/main.c:27]: Is this needed? As we are only calling the Wi-Fi mgmt ext API?

File: samples/wifi/sta/src/main.c [hunk 3]
```diff
@@ -207,56 +212,18 @@ static void net_mgmt_event_handler(struct net_mgmt_event_callback *cb,
...
 	context.connect_result = false;
-	__wifi_args_to_params(&cnx_params);
 
-	if (net_mgmt(NET_REQUEST_WIFI_CONNECT, iface,
-		     &cnx_params, sizeof(struct wifi_connect_req_params))) {
+	/* Apply stored wifi credential */
+	ret = net_mgmt(NET_REQUEST_WIFI_CONNECT_STORED, iface, NULL, 0);
+	if (ret) {
 		LOG_ERR("Connection request failed");
-
```
Comment (krish2718) [inline samples/wifi/sta/src/main.c:259]: unnecessary whitespace

File: samples/wifi/sta/src/main.c (hunk 2 above)
Comment (D-Triveni) [reply samples/wifi/sta/src/main.c:27]: Not needed.

**Review by bama-nordic — APPROVED** [review]

**Comment by D-Triveni** [comment]
No longer needed.
